File: feature/keyboard/src/main/java/net/twisterrob/colorfilters/android/keyboard/KeyboardHandlerFactory.java
package net.twisterrob.colorfilters.android.keyboard;

import android.inputmethodservice.KeyboardView;
import android.support.annotation.NonNull;
import android.view.Window;

public class KeyboardHandlerFactory {

	public KeyboardHandler create(
			@NonNull KeyboardMode mode, @NonNull Window window, @NonNull KeyboardView keyboardView) {
		switch (mode) {
			case Hex:
				return new HexKeyboardHandler(window, keyboardView);
			case FloatNav:
				return new FloatNavKeyboardHandler(window, keyboardView);
			case Float:
				return new FloatKeyboardHandler(window, keyboardView);
			case NATIVE:
				return new NullKeyboardHandler(window, keyboardView);
			default:
				throw new UnsupportedOperationException("Cannot create keyboard for " + mode);
		}
	}
}

File: feature/keyboard/src/main/kotlin/net/twisterrob/colorfilters/android/keyboard/KeyboardHandlerFactory.kt
package net.twisterrob.colorfilters.android.keyboard

import android.view.Window
import net.twisterrob.colorfilters.android.keyboard.KeyboardMode.Float
import net.twisterrob.colorfilters.android.keyboard.KeyboardMode.FloatNav
import net.twisterrob.colorfilters.android.keyboard.KeyboardMode.Hex
import net.twisterrob.colorfilters.android.keyboard.KeyboardMode.NATIVE

class KeyboardHandlerFactory {

	fun create(
		mode: KeyboardMode,
		window: Window,
		@Suppress("DEPRECATION")
		keyboardView: android.inputmethodservice.KeyboardView
	): KeyboardHandler =
		when (mode) {
			Hex -> HexKeyboardHandler(window, keyboardView)
			FloatNav -> FloatNavKeyboardHandler(window, keyboardView)
			Float -> FloatKeyboardHandler(window, keyboardView)
			NATIVE -> NullKeyboardHandler(window, keyboardView)
		}
}
